src/Esprima/ParserOptions.cs

```diff
@@ -34,17 +37,27 @@ public record class ParserOptions
     /// <summary>
     /// Gets or sets the <see cref="ErrorHandler"/> to use, defaults to <see cref="ErrorHandler.Default"/>.
     /// </summary>
-    public ErrorHandler ErrorHandler { get => ScannerOptions._errorHandler; init => ScannerOptions._errorHandler = value; }
+    public ErrorHandler ErrorHandler { get => _scannerOptions._errorHandler; init => _scannerOptions._errorHandler = value; }
 
     /// <summary>
     /// Gets or sets whether the Regular Expression syntax should be converted to a .NET compatible one, defaults to <see langword="true"/>.
     /// </summary>
-    public bool AdaptRegexp { get => ScannerOptions._adaptRegexp; init => ScannerOptions._adaptRegexp = value; }
+    [Obsolete($"This property is planned to be removed from the next stable version. Please use the {nameof(RegExpParseMode)} property instead.")]
```

Comment:
This one is not entirely clear to me. Instead of readme, didn't you mean release notes? That would make more sense. However, I'd also keep the obsolete attribute so early adopters can prepare their code for this change.

Reply:
Yes, I meant any documentation that would accompany the final 3.x release. I was just wondering how many RCs there will be before the next stable - but would make sense to have at least one to to prepare people who live on the edge - I just wouldn't want obsolete attributes in final 3.0 as we've been doing so much cleanup already and it's again another TODO for 4.x 😉 .

Reply:
>  I just wouldn't want obsolete attributes in final 3.0
Fully agreed. I also have it in mind to remove the deprecated stuff from the final v3.x.

>  I was just wondering how many RCs there will be before the next stable
Hopefully, we won't need more than one or two RCs. AFAICS, [this one](https://github.com/sebastienros/esprima-dotnet/pull/381) can be merged soon. Apart from that, I have some minor polishing left.

BTW, I added a BC section to the top comment to help us with writing the release notes.